Add federated API documentation (#104)
OpenTermsArchive/docs: content/federated-api.md

Reviewer: Since we create a new page, I'd be in favour of paying back a tiny bit of our debt and creating an “API” category into which we'd move some of the main page content, with the following layout:

- API
  - Node.js
  - HTTP
  - Federated

The beta markers would then be added in the title, like in the Federated API page.

---
title: "Federated API"
weight: 7
---

# Federated API [Beta]

> Open Terms Archive is a decentralised system that tracks collections of services' terms across multiple servers. Each collection operates its [own API](/#api), and the federated API simplify searching across these collections, fostering collaboration with external applications. This provides a simplified method for discovering service terms within the Open Terms Archive decentralized system.

Reviewer: ```suggestion
Open Terms Archive is a decentralised system that tracks collections of services' terms across multiple servers. Each collection operates its [own API](/#api), and the federated API unifies search and discovery across collections, fostering collaboration with external applications.
```

## Documentation

> The Federated Web API exposes JSON data over HTTP(S). The [documentation](http://51.89.136.45/v1/docs/) is publicly available for exploration. That endpoint exposes both the [OpenAPI](https://swagger.io/specification/) specification if the requested `Content-Type` is JSON, and a Swagger UI for visual and interactive documentation otherwise.

Reviewer: > Not sure if mentioning SSL without mentioning it makes things more complicated.
```suggestion
The Federated Web API exposes JSON data over HTTP. Its [documentation](http://51.89.136.45/v1/docs/) is provided in a dedicated, interactive interface.

That endpoint exposes both the [OpenAPI](https://swagger.io/specification/) specification if the requested `Content-Type` is JSON, and a Swagger UI for visual and interactive documentation otherwise.
```

> 

Reviewer: ```suggestion

## Beta

This API is offered as a preview, based on a first use case [defined](https://github.com/OpenTermsArchive/engine/issues/1016) with partner [ToS;DR](https://tosdr.org). Unexpected problems or missing functionality may arise. Please provide feedback through [issues](https://github.com/OpenTermsArchive/federated-api/issues) in the dedicated repository.

```

## Source code

> The codebase for this Federated API is available on [https://github.com/OpenTermsArchive/federated-api](https://github.com/OpenTermsArchive/federated-api).

Reviewer: ```suggestion
The codebase for the Federated API is available on [`github.com/OpenTermsArchive/federated-api`](https://github.com/OpenTermsArchive/federated-api).
```
